packages/backend/src/auth/account/account.ts

```diff
@@ -0,0 +1,127 @@
+    embedToken: string,
+    organization: Pick<Organization, 'organizationUuid' | 'name'>,
+): string {
+    return (
+        decodedToken.user?.externalId ||
+        `anonymous-jwt::${organization.organizationUuid}_${embedToken}`
+    );
+}
+
+function createAccount(account: Partial<SessionAccount>): SessionAccount;
+function createAccount(account: Partial<ExternalAccount>): ExternalAccount;
+function createAccount<T extends Account>(account: Partial<T>): T {
+    if (!account.user?.ability || !account.user?.abilityRules) {
+        throw new ForbiddenError(
+            'User ability and abilityRules are required for permissions',
+        );
+    }
+
+    return {
+        ...account,
+        isAuthenticated: () => !!account.authentication && !!account.user?.id,
+        isKnownUser: () => account.user?.type === 'known',
+        isAnonymousUser: () => account.user?.type !== 'known',
+        isSessionUser: () => account.authentication?.type === 'session',
+        isJwtUser: () => account.authentication?.type === 'jwt',
+    } as T;
+}
+
+export function fromJwt({
+    decodedToken,
+    organization,
+    source,
+    dashboardUuid,
+    userAttributes,
+}: {
+    decodedToken: CreateEmbedJwt;
+    organization: Embed['organization'];
+    source: string;
+    dashboardUuid: string;
+    userAttributes: AccountAccessControls;
+}) {
+    const builder = new AbilityBuilder<MemberAbility>(Ability);
+    applyEmbeddedAbility(decodedToken, dashboardUuid, organization, builder);
+    const abilities = builder.build();
+
+    return createAccount({
+        authentication: {
+            type: 'jwt',
+            data: decodedToken,
+            source,
+        },
+        organization,
+        access: {
+            dashboardId: dashboardUuid,
+            filtering: decodedToken.content.dashboardFiltersInteractivity,
+            controls: userAttributes,
+        },
+        // Create the fields we're able to set from the JWT
+        user: {
+            id: getExternalId(decodedToken, source, organization),
+            type: 'external',
+            ability: abilities,
+            abilityRules: abilities.rules,
+            email: decodedToken.user?.email,
+            isActive: true,
+        },
+    });
+}
+
+export function fromSession(
+    sessionUser: SessionUser,
+    source: string = '',
+): SessionAccount {
+    const {
+        organizationCreatedAt,
+        organizationName,
+        organizationUuid,
+        ...user
+    } = sessionUser;
+
+    if (!organizationUuid || !organizationName || !organizationCreatedAt) {
+        const missing: string[] = [];
+        if (!organizationUuid) missing.push('organizationUuid');
+        if (!organizationName) missing.push('organizationName');
+        if (!organizationCreatedAt) missing.push('organizationCreatedAt');
+
+        throw new ForbiddenError(
+            `Missing required organization fields: ${missing.join(', ')}.`,
+        );
+    }
```

Comment:
Replaces all the checks for isUserWithOrg throughout the app:
https://github.com/lightdash/lightdash/blob/8e03c6bc9cda1aa41a57730ca6dd000e8a1ff8c9/packages/common/src/types/user.ts#L65-L71

Reply:
@ZeRego added `isUserWithOrg` - do you remember why this was needed?

Reply:
I'm going to remove this check from Oliver's [Slack context](https://lightdash.slack.com/archives/C08L4N6GU1J/p1752751409411389?thread_ts=1752744894.132299&cid=C08L4N6GU1J).

@ZeRego does CASL not catch what this check is looking for?